src/scripts/main.js

```diff
@@ -1 +1,105 @@
 'use strict';
+
+const firstPromise = new Promise((resolve, reject) => {
+  let isResolved = false;
+
+  const clickHandler = () => {
+    if (!isResolved) {
+      isResolved = true;
+      clearTimeout(timer);
+      document.removeEventListener('click', clickHandler);
+
+      resolve('First promise was resolved');
+    }
+  };
+
+  document.addEventListener('click', clickHandler);
+
+  const timer = setTimeout(() => {
+    if (!isResolved) {
+      isResolved = true;
+      document.removeEventListener('click', clickHandler);
+
+      reject(new Error('First promise was rejected'));
```

Comment:
When rejecting a promise, it's better to provide an error object with a meaningful message, which you've done correctly. However, consider including additional context or information that could help debug the promise rejection if needed.